Products/ZenEvents/EventClass.py

```diff
@@ -240,11 +240,11 @@ def getInstances(self):
     def nextSequenceNumber(self, key):
         """Get next sequence number for instance.
         """
-        idx = 0
-        insts = self.find(key)
-        if len(insts) > 0:
-            idx = insts[-1].sequence + 1
-        return idx
+        instances = self.find(key)
+        if len(instances) > 0:
+            return max([i.sequence for i in instances])+1
+        else:
+            return 0
```

Comment:
The `else:` is redundant when there's no work to do when the `if` statement is false. Also, the list comprehension given to `max` should be a generator comprehension.  This block of code can be written more succinctly as
```
return max(i.sequence for i instances) + 1 if len(instances) else 0
```
or
```
if len(instances):
    return max(i.sequence for i in instances) + 1
return 0
```
Note: I used `len(instances)` as the test condition because `len(seq)` and `len(seq) > 0` are equivalent in boolean contexts (such as the condition of an `if` statement).

Reply:
Good point! :+1: